core/modules/rproc/InfileMerger.cc

```diff
@@ -502,9 +492,13 @@ bool InfileMerger::makeResultsTableForQuery(query::SelectStmt const& stmt) {
     }
     _addJobIdColumnToSchema(schema);
     std::string createStmt = sql::formCreateTable(_mergeTable, schema);
-    //createStmt += " ENGINE=MyISAM"; // &&&
-    //createStmt += " ENGINE=MEMORY"; // &&&
-    createStmt += " ENGINE=InnoDB"; // &&&
+    if (_dbEngine == INNODB) {
+        createStmt += " ENGINE=InnoDB";
+    } else if (_dbEngine == MEMORY) {
+        createStmt += " ENGINE=MEMORY";
+    } else { // MYISAM
+        createStmt += " ENGINE=MyISAM";
+    }
```

Comment:
This has to be `switch(_dbEngine) { ...` and an exception `throw` for the rest (unknown table type).